daemon/chaintopology.c

```diff
@@ -173,6 +173,9 @@ static void connect_block(struct lightningd_state *dstate,
 			add_tx_to_block(b, &txid);
 	}
 	b->full_txs = tal_free(b->full_txs);
+
+	/* Tell peers about new block. */
+	peers_new_block(dstate, b->height);
```

Comment:
We risk entering callback hell if we do this, however it's trivial to generalize later as you say.

I'm not aware of a publish/subscribe implementation, but I wonder if boutique would be simpler than producing one which handled all the cases.